libs/service/api_service.cpp

```diff
@@ -17,7 +16,181 @@ extern "C"
 }
 #include "Verifier.h"
 #include "verifier_service.h"
-#include "windows_helpers.hpp"
+
+#define MAX_CODE_SIZE (32 * 1024) // 32 KB
+
+static ebpf_result_t
+_build_helper_id_to_address_map(
+    ebpf_handle_t program_handle, ebpf_code_buffer_t& byte_code, std::map<uint32_t, uint64_t>& helper_id_to_adddress)
+{
+    ebpf_inst* instructions = reinterpret_cast<ebpf_inst*>(byte_code.data());
+    for (size_t index = 0; index < byte_code.size() / sizeof(ebpf_inst); index++) {
+        ebpf_inst& instruction = instructions[index];
+        if (instruction.opcode != INST_OP_CALL) {
+            continue;
+        }
+        helper_id_to_adddress[instruction.imm] = 0;
+    }
+
+    if (helper_id_to_adddress.size() == 0)
+        return EBPF_SUCCESS;
+
+    ebpf_protocol_buffer_t request_buffer(
+        offsetof(ebpf_operation_resolve_helper_request_t, helper_id) + sizeof(uint32_t) * helper_id_to_adddress.size());
+
+    ebpf_protocol_buffer_t reply_buffer(
+        offsetof(ebpf_operation_resolve_helper_reply_t, address) + sizeof(uint64_t) * helper_id_to_adddress.size());
+
+    auto request = reinterpret_cast<ebpf_operation_resolve_helper_request_t*>(request_buffer.data());
+    auto reply = reinterpret_cast<ebpf_operation_resolve_helper_reply_t*>(reply_buffer.data());
+    request->header.id = ebpf_operation_id_t::EBPF_OPERATION_RESOLVE_HELPER;
+    request->header.length = static_cast<uint16_t>(request_buffer.size());
+    request->program_handle = reinterpret_cast<uint64_t>(program_handle);
+
+    size_t index = 0;
+    for (const auto& helper : helper_id_to_adddress) {
+        request->helper_id[index++] = helper.first;
```

Comment:
```suggestion
    for (const auto& [helper_id, address]: helper_id_to_adddress) {
        request->helper_id[index++] = helper_id;
```
use structured bindings instead of .first and .second